pkg/network/protocols/postgres/telemetry.go

```diff
@@ -0,0 +1,104 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package postgres
+
+import (
+	"fmt"
+
+	libtelemetry "github.com/DataDog/datadog-agent/pkg/network/protocols/telemetry"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	numberOfBuckets        = 10
+	bucketRange            = 15
+	belowBufferBucketCount = 3
+)
+
+// Telemetry is a struct to hold the telemetry for the postgres protocol
+type Telemetry struct {
+	metricGroup *libtelemetry.MetricGroup
+
+	// queryLengthBuckets holds the counters for the different buckets of by the query length quires
+	queryLengthBuckets [numberOfBuckets]*libtelemetry.Counter
+	// failedTableNameExtraction holds the counter for the failed table name extraction
+	failedTableNameExtraction *libtelemetry.Counter
+	// failedOperationExtraction holds the counter for the failed operation extraction
+	failedOperationExtraction *libtelemetry.Counter
+}
+
+// createQueryLengthBuckets initializes the query length buckets
+// Bucket 1: <= 34   query length
+// Bucket 2: 35 - 49 query length
+// Bucket 3: 50 - 64 query length
+// Bucket 4: 65 - 79 query length
+// Bucket 5: 80 - 94 query length
+// Bucket 6: 95 - 109 query length
+// Bucket 7: 110 - 124 query length
+// Bucket 8: 125 - 139 query length
+// Bucket 9: 140 - 154 query length
+// Bucket 10: >= 155 query length
+func createQueryLengthBuckets(metricGroup *libtelemetry.MetricGroup) [numberOfBuckets]*libtelemetry.Counter {
+	var buckets [numberOfBuckets]*libtelemetry.Counter
+	for i := 0; i < numberOfBuckets; i++ {
+		buckets[i] = metricGroup.NewCounter("query_length_bucket"+fmt.Sprint(i+1), libtelemetry.OptStatsd)
+	}
+	return buckets
+}
+
+// NewTelemetry creates a new Telemetry
+func NewTelemetry() *Telemetry {
+	metricGroup := libtelemetry.NewMetricGroup("usm.postgres")
+
+	return &Telemetry{
+		metricGroup:               metricGroup,
+		queryLengthBuckets:        createQueryLengthBuckets(metricGroup),
+		failedTableNameExtraction: metricGroup.NewCounter("failed_table_name_extraction", libtelemetry.OptStatsd),
+		failedOperationExtraction: metricGroup.NewCounter("failed_operation_extraction", libtelemetry.OptStatsd),
+	}
+}
+
+// getBucketIndex returns the index of the bucket for the given query size
+func getBucketIndex(querySize int) int {
+	// Add 1 to the start size to include BufferSize (64) as the last value of bucket 3, resulting in the following order:
+	// The first three buckets will include sizes below the current buffer size,
+	// and the rest will include sizes equal to or above the buffer size.
+	startSize := BufferSize - belowBufferBucketCount*bucketRange + 1
```

Comment:
Should be a constant (or to be calculated only once)
and rename to firstBucketLowerBoundary or something clear

Reply:
also, the calculation is wrong

Reply:
should be

lowerBound := BufferSize - (belowBufferBucketCount - 1) * bucketSize